datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1947,6 +1858,53 @@ fn has_common_conjunction(lhs: &Expr, rhs: &Expr) -> bool {
     iter_conjunction(rhs).any(|e| lhs_set.contains(&e) && !e.is_volatile())
 }
 
+fn binary_op_null_on_null(op: Operator) -> bool {
```

Comment:
I recommend making this a function on `Operator` so it is more discoverable -- similar to https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.Operator.html#method.is_numerical_operators

Reply:
done!